minorminer/utils/zephyr/qfloor.py

```diff
@@ -0,0 +1,402 @@
+# Copyright 2025 D-Wave
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+# ================================================================================================
+
+
+from __future__ import annotations
+
+from collections import defaultdict, namedtuple
+from itertools import product
+from functools import cached_property
+from typing import Callable, Iterable, Iterator
+
+from minorminer.utils.zephyr.coordinate_systems import *
+from minorminer.utils.zephyr.node_edge import ZEdge, ZNode, ZShape, EdgeKind
+from minorminer.utils.zephyr.plane_shift import PlaneShift
+
+Dim_2D = namedtuple("Dim_2D", ["Lx", "Ly"])
+UWJ = namedtuple("UWJ", ["u", "w", "j"])
+ZSE = namedtuple("ZSE", ["z_start", "z_end"])
+
+
+class QuoTile:
+    """Initializes a 'QuoTile' object with zns.
```

Comment:
Better to be explicit.
```suggestion
    """Initializes a 'QuoTile' object with ``ZNode``.
```

Reply:
Done in 659c815.